src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,29 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    WSingletonContainer::defineSingleton(child_group->objectName(),
+                                         child_group);
+}
+
+QWidget* LegacySkinParser::parseSingletonContainer(QDomElement node) {
+    if (!node.hasAttribute("objectName")) {
+        qWarning() << "Need objectName attribute for Singleton tag.";
+        return new QWidget();
+    }
+    QString objectName = node.attribute("objectName");
+    WSingletonContainer* pContainer =
+            WSingletonContainer::getSingleton(objectName, m_pParent);
+    if (pContainer == NULL) {
+        qWarning() << "ERROR: Singleton" << objectName
+                   << "not found, probably about to crash.";
+    }
+    commonWidgetSetup(node, pContainer);
```

Comment:
note that there is in fact a new widget -- the SingletonContainer itself is a WidgetGroup that needs common setup.